modules/system-disaster-recovery/src/test/java/org/apache/ignite/internal/disaster/system/SystemDisasterRecoveryManagerImplTest.java

```diff
@@ -0,0 +1,490 @@
+                manager.resetCluster(List.of(thisNodeName)),
+                ClusterResetException.class,
+                10, SECONDS
+        );
+        assertThat(ex.getMessage(), is("Node is not initialized and cannot serve as a cluster reset conductor."));
+    }
+
+    private void putClusterState() {
+        clusterStateStorageManager.putClusterState(usualClusterState);
+    }
+
+    @Test
+    void resetClusterRequiresNodeToHaveSavedClusterName() {
+        when(topologyService.allMembers()).thenReturn(List.of(thisNode));
+        putClusterState();
+        makeNodeInitialized();
+
+        ClusterResetException ex = assertWillThrow(
+                manager.resetCluster(List.of(thisNodeName)),
+                ClusterResetException.class,
+                10, SECONDS
+        );
+        assertThat(ex.getMessage(), is("Node does not have cluster name saved and cannot serve as a cluster reset conductor."));
+    }
+
+    private void makeNodeInitialized() {
+        vaultManager.put(new ByteArray(NODE_INITIALIZED_VAULT_KEY), new byte[]{1});
+    }
+
+    @Test
+    void resetClusterSendsMessages() {
+        when(topologyService.allMembers()).thenReturn(List.of(thisNode, node2, node3));
+        prepareNodeStateForClusterReset();
+
+        when(messagingService.invoke(any(ClusterNode.class), any(), anyLong()))
+                .thenReturn(completedFuture(successResponseMessage));
+
+        CompletableFuture<Void> future = manager.resetCluster(List.of(thisNodeName, node2.name()));
+        assertThat(future, willCompleteSuccessfully());
+
+        verify(messagingService).invoke(eq(thisNode), resetClusterMessageCaptor.capture(), anyLong());
+        ResetClusterMessage messageToSelf = resetClusterMessageCaptor.getValue();
+        assertThatResetClusterMessageIsAsExpected(messageToSelf);
+
+        verify(messagingService).invoke(eq(node2), resetClusterMessageCaptor.capture(), anyLong());
+        ResetClusterMessage messageToOtherNewCmgNode = resetClusterMessageCaptor.getValue();
+        assertThatResetClusterMessageIsAsExpected(messageToOtherNewCmgNode);
+
+        verify(messagingService).invoke(eq(node3), resetClusterMessageCaptor.capture(), anyLong());
+        ResetClusterMessage messageToOtherNonCmgNode = resetClusterMessageCaptor.getValue();
+        assertThatResetClusterMessageIsAsExpected(messageToOtherNonCmgNode);
+
+        assertThat(messageToSelf.clusterId(), is(messageToOtherNewCmgNode.clusterId()));
+        assertThat(messageToSelf.clusterId(), is(messageToOtherNonCmgNode.clusterId()));
+    }
+
+    private void prepareNodeStateForClusterReset() {
+        putClusterNameToVault();
+        makeNodeInitialized();
+        putClusterState();
+    }
+
+    private void putClusterNameToVault() {
+        vaultManager.put(new ByteArray(CLUSTER_NAME_VAULT_KEY), CLUSTER_NAME.getBytes(UTF_8));
+    }
+
+    private void assertThatResetClusterMessageIsAsExpected(ResetClusterMessage message) {
```

Comment:
Isn't it enough to call `assertEquals`?

Reply:
What's a message content? Why isn't it covered by the `equals` method of the message?
If we don't properly override `equals` for messages then that's fine, just say so

Reply:
I meant comparing one message with another, but I see that we never construct another message for comparison. I guess the discussion might be closed, I misinterpreted the code somehow